planner/aamwriter/src/main/java/eu/seaclouds/platform/planner/aamwriter/modeldesigner/DGraph.java

```diff
@@ -122,13 +127,27 @@ public DNode getNode(String nodeName) {
         return result;
     }
 
+    private DNode calculateSourceNode() {
+        DNode result = DNode.NOT_FOUND;
+        
+        for (DNode node : nodes) {
+            List<DLink> links = getLinksTo(node);
+            if (links.size() == 0) {
+                result = node;
+                break;
+            }
+        }
+        return result;
+    }
+    

```

Comment:
Just a comment: Here you are assuming that every application has exactly one node with 0 incoming links. Is this enforced by the GUI? I mean is the GUI forcing users to develop application that:
- cannot have all nodes with incoming dependecies (i.e. are all application directed acyclic graphs?), and
- cannot have two separate, independent stacks?

If not, you may encounter issues with this code.